### ISSUE TITLE
Question on sinusoidal equations

### ISSUE THREAD

**Issue by ziyinyuan** [dynamicslab/pysindy#212]
So I want to try a basic x_dot=cos(t) function, but I am having problems with that in pysindy.

dt=0.001
t=np.arange(0,10,dt)

x=np.cos(t)

---------------------

model=ps.SINDy(feature_library=ps.FourierLibrary(n_frequencies=1),
               optimizer=ps.STLSQ(threshold=0.01),
               differentiation_method=ps.FiniteDifference(order=2),
               feature_names=["x"])
model.fit(x,t=t)
model.print()

---------------------

x0_train=[1]
t_train=np.arange(0,10,0.1)
x_sim=model.simulate(x0_train,t_train)

plt.plot(t,x,"r",label="Exact Solution",linewidth=4)
plt.plot(t_train,x_sim,"k--",label="SINDy model",linewidth=3)
plt.legend()

-------------------

The x_sim is not showing a sinusoidal plot, it is just showing a curve.
Please correct me if I did something wrong here.

**Comment by ziyinyuan** [comment]
Thank you for pointing out the error. Would you suggest using a custom lib in this case? Or Fourier lib can do the job?

**Comment by akaptano** [comment]
The custom library will not fix this issue, since the custom functions still are applied to the state variable x and not the time variable. You will need to use a GeneralizedLibrary with a Fourier Library + specifying to use the time variable instead of the state variable (see examples notebooks 1 and 15). Alternatively, you can simply note that if x = cos(t), x' = -sin(t) = - sqrt(1- x^2). 

Best,
Alan

**Comment by ziyinyuan** [comment]
Thank you for your patience Alan. 
I am not sure how to specify the time variable except using "feature_names=['t']", but this would change the formula to t'=...(t) instead of x'=...(t). 
I am also not sure what to specify in generalized lib. I am assuming I don't need any tensor array since I am only using 1 lib which is the Fourier lib, and probably don't need any inputs_per_lib because there's only 1 input in this case?

And alternatively, would polynomial library able to get x' = - sqrt(1- x^2)?

**Comment by ziyinyuan** [comment]
I also tried inputs_per_lib for one input only, but it keep giving me errors.

**Comment by akaptano** [comment]
please let us know what the errors were. You should be able to use inputs_per_lib to specify that the library should build terms using the "t" variable instead of the "x" variable.

**Comment by ziyinyuan** [comment]
I use:
----------------
inputs_per_lib=[[1,1]]

generalized_library=ps.GeneralizedLibrary(
    [ps.FourierLibrary()],
    inputs_per_library=inputs_per_lib
)
------------
Then error shows: 'list' object has no attribute 'ndim'.
I was thinking to call this later on:
----------
model=ps.SINDy(
feature_names=['x','t']
)
--------------

**Comment by Jacob-Stevens-Haas** [comment]
### Solution

This:
```python
import pysindy as ps
import numpy as np

dt=0.001
t=np.arange(0,10,dt)

x=np.cos(t)

lib = ps.GeneralizedLibrary(
    libraries=[ps.FourierLibrary(n_frequencies=1)],
    inputs_per_library=np.array([[0, 1]])
)

model=ps.SINDy(
    feature_library=lib,
    optimizer=ps.STLSQ(threshold=0.01),
    differentiation_method=ps.FiniteDifference(order=2),
    feature_names=["x", "t"])
model.fit(x,t=t, u=t)
model.print()
```
correctly prints:
```
(x)' = -1.000 sin(1 t)
```

### Explanation
When you want variables (e.g. $t$) to affect $x'$, but you do not want to fit models for them, you can treat them as control inputs `u`.  As @akaptano mentioned, the line `inputs_per_library=np.array([[0, 1]])` tells the `GeneralizedLibrary` that it will only give the second variable to the `FourierLibrary`.  When you call `model.fit(x, t=t, u=t)`, pysindy appends the control input `u` to the independent variable `x` before sending the data to the `GeneralizedLibrary`
